src/main/java/de/numcodex/sq2cql/model/Mapping.java

```
@@ -7,53 +7,38 @@
 
 import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
 
+import static java.util.Objects.requireNonNull;
+
 /**
  * @author Alexander Kiel
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
-public final class Mapping {
-
-    private final TermCode key;
-    private final String resourceType;
-    private final String valueFhirPath;
-    private final List<Modifier> fixedCriteria;
-
-    private Mapping(TermCode key, String resourceType, String valueFhirPath, List<Modifier> fixedCriteria) {
-        this.key = Objects.requireNonNull(key);
-        this.resourceType = Objects.requireNonNull(resourceType);
-        this.valueFhirPath = valueFhirPath;
-        this.fixedCriteria = Objects.requireNonNull(fixedCriteria);
+public record Mapping(TermCode key, String resourceType, String valueFhirPath, List<Modifier> fixedCriteria) {
+
+    public Mapping {
+        requireNonNull(key);
+        requireNonNull(resourceType);
+        requireNonNull(valueFhirPath);
+        requireNonNull(fixedCriteria);
     }
 
     public static Mapping of(TermCode concept, String resourceType) {
-        return new Mapping(Objects.requireNonNull(concept), Objects.requireNonNull(resourceType), null, List.of());
+        return new Mapping(requireNonNull(concept), requireNonNull(resourceType), "value", List.of());
     }
 
     @JsonCreator
     public static Mapping of(@JsonProperty("key") TermCode key,
                              @JsonProperty("fhirResourceType") String resourceType,
                              @JsonProperty("valueFhirPath") String valueFhirPath,
                              @JsonProperty("fixedCriteria") Modifier... fixedCriteria) {
-        return new Mapping(Objects.requireNonNull(key), Objects.requireNonNull(resourceType), valueFhirPath,
+        return new Mapping(requireNonNull(key), requireNonNull(resourceType),
+                valueFhirPath == null ? "value" : valueFhirPath,
```

We should not default to "value". If no valueFhirPath is given the default is not value. This could prevent us from failing fast.

---

Previously I had "value" hard coded. We have to update all mappings to make that work without this default.